app/hotels/src/map/HotelSnaps.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import * as React from 'react';
+import { StyleSheet, Dimensions, View } from 'react-native';
+import Carousel from 'react-native-snap-carousel';
+import {
+  HotelSnap,
+  Address,
+  type OnDimensionsChange,
+} from '@kiwicom/react-native-app-common';
+
+type Props = {|
+  items: Object[], // todo
+  selectedIndex: number,
+  onSnapToItem: (index: number) => void,
+|};
+
+type State = {|
+  availableWidth: number,
+|};
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#fff',
+    height: 130,
+  },
+  slider: {
+    padding: 10,
+  },
+});
+
+const SNAP_WIDTH = 0.8;
+
+class HotelSnaps extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      availableWidth: Dimensions.get('screen').width,
+    };
+  }
+
+  componentDidMount = () => {
+    Dimensions.addEventListener('change', this.onDimensionsChanged);
+  };
+
+  componentWillUnmount = () => {
+    Dimensions.removeEventListener('change', this.onDimensionsChanged);
+  };
+
+  onDimensionsChanged = ({ screen: { width } }: OnDimensionsChange) => {
+    this.setState({ availableWidth: width });
```

Comment:
Why is this needed? Cannot we just use "width" from `Dimensions`? The trick is that you have to use it inside of render function so it gets updated every time you rerender (this is what you want anyway).

Reply:
IMHO component is not re-rendered because the device has rotated, I even tried that, that's why I added lisneter, but I may miss something you already know...

Reply:
I did it here: https://github.com/kiwicom/react-native-app/blob/master/app/hotels/src/gallery/PhotosStripe.js#L23 (just to fix exactly the same issue)

Reply:
It's described here: https://facebook.github.io/react-native/docs/dimensions.html#get But now I really don't know why does it work... ❓

Reply:
Ok I bet it's about `onLayout` change in the `Swiper` so it doesn't work exactly how I thought... 🙂